src/shared/read-agents.c

```diff
@@ -1343,3 +1343,31 @@ char **get_agents(int flag)
     closedir(dp);
     return (f_files);
 }
+
+int check_discon_keepalvie(char *last_keepalive) {
+    struct tm st_time, *st_now;
+    time_t t_unified;
+    time_t t_now;
+
+    if (last_keepalive) {
+        if(!strptime(last_keepalive, "%Y-%m-%d %H:%M:%S", &st_time)) {
+            merror("at check_discon_keepalvie(): strptime().");
+            return 0;
+        }
+        st_time.tm_isdst = -1;
```

Comment:
I suggest you fully initialize the variable. https://github.com/wazuh/wazuh/blob/0d73e273527102051a25b66ca607b68f79740b73/src/os_csyslogd/alert.c#L394-L399